packages/wrangler/src/publish/publish.ts

```diff
@@ -647,6 +648,80 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 					"Content-Type": "application/json",
 				},
 			});
+		} else if (routes.length !== 0) {
+			// this is kinda unreadable, huh.
+			//
+			// basically it's just like, if you get to this point
+			// you're trying to deploy a worker to a custom
+			// domain that's already bound to another worker.
+			// so this big array traversal thing is just about
+			// finding workers that have bindings to the routes
+			// you're trying to deploy to.
+			//
+			// the logic is kinda similar (read: duplicated) from publishRoutesFallback,
+			// except here we know we have a good API token or whatever so we don't need
+			// to bother with all the error handling tomfoolery.
+			const routesWithOtherBindings = (
+				await Promise.all(
+					routes.map(async (route) => {
+						const zone = await getZoneForRoute(route);
+						if (!zone) {
+							return [];
+						}
+
+						const routePattern =
+							typeof route === "string" ? route : route.pattern;
+
+						return (
+							await fetchListResult<{
+								pattern: string;
+								script: string;
+							}>(`/zones/${zone.id}/workers/routes`)
+						).filter(
+							({ pattern, script }) =>
+								pattern === routePattern && script !== scriptName
+						);
+					})
+				)
+			)
+				.flat()
+				.reduce((scriptPatternMap, { pattern, script }) => {
```

Comment:
Does this need to be a reduce? I feel like a for loop would be clearer (especially with the chain of functions)?

Reply:
💯

Reply:
Sure